capi/src/lib.rs

```diff
@@ -0,0 +1,195 @@
+#![allow(non_camel_case_types)]
+
+extern crate libc;
+extern crate telemetry;
+
+use std::ptr;
+use std::ffi::{CStr, CString};
+use std::mem;
+use std::str;
+use std::sync::mpsc::channel;
+use telemetry::{Subset, Service, SerializationFormat};
+use telemetry::plain::{Histogram, Flag, Count};
+
+pub struct telemetry_t {
+    service: Service,
+    flags: Vec<*mut flag_t>,
+    counts: Vec<*mut count_t>,
+}
+
+impl telemetry_t {
+    fn new(is_active: bool) -> telemetry_t {
+        telemetry_t {
+            flags: vec![],
+            counts: vec![],
+            service: Service::new(is_active),
+        }
+    }
+}
+
+impl Drop for telemetry_t {
+    fn drop(&mut self) {
+        unsafe {
+            for flag in &self.flags {
+                free_flag(*flag);
+            }
+
+            for count in &self.counts {
+                free_count(*count);
+            }
+        }
+    }
+}
+
+trait CHistogram {
+    fn new(name: &str, telemetry: &Service) -> Self;
```

Comment:
This signature may not work for all types of histograms. For instance, `Linear` (and, in the future, `Exponential`) need a number of buckets.